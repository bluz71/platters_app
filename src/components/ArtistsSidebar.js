import React from 'react'
import { Col, ListGroup, ListGroupItem } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import '../styles/ArtistsSidebar.css'

const renderAlbums = albums =>
  albums.map(album => (
    <ListGroupItem key={album.id} className='album'>
      <h5>
        <Link to={`/artist/${album.artist_slug}/album/${album.album_slug}`}>
          {album.title}
        </Link>
      </h5>
      <h5>
        by <Link to={`/artist/${album.artist_slug}`}>{album.artist}</Link>
      </h5>
      <Link to={`/artist/${album.artist_slug}/album/${album.album_slug}`}>
        <img
          className='img-responsive center-block'
          src={album.cover_url}
          alt={album.title}
        />
      </Link>
    </ListGroupItem>
  ))

const renderComments = comments =>
  comments.map(comment => (
    <ListGroupItem key={comment.id} className='comment'>
      <h5 dangerouslySetInnerHTML={{ __html: `${comment.created_at} by` }} />
      <h5>
        <Link to={`/comments/${comment.user_slug}`}>{comment.user_name}</Link>
      </h5>
      <Link to={`/comments/${comment.user_slug}`}>
        <img
          className='img-responsive center-block'
          src={comment.gravatar_url}
          alt={comment.user_name}
        />
      </Link>
      <h5>
        in{' '}
        <Link
          to={{
            pathname: comment.path,
            state: { scrollToComments: true }
          }}
        >
          {comment.name}
        </Link>
      </h5>
    </ListGroupItem>
  ))

const ArtistsSidebar = ({ mostRecentAlbums, mostRecentComments }) => {
  if (mostRecentAlbums.length === 0) {
    return <aside />
  }

  return (
    <aside className='ArtistsSidebar'>
      <Col md={2}>
        <section className='panel panel-default'>
          <div className='panel-heading'>
            <h4 className='panel-title'>New Albums</h4>
          </div>
          <ListGroup>{renderAlbums(mostRecentAlbums)}</ListGroup>
        </section>
        <section className='panel panel-default'>
          <div className='panel-heading'>
            <h4 className='panel-title'>New Comments</h4>
          </div>
          <ListGroup>{renderComments(mostRecentComments)}</ListGroup>
        </section>
      </Col>
    </aside>
  )
}

export default ArtistsSidebar
