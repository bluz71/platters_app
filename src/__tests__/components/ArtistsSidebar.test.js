import React from 'react'
import { shallow } from 'enzyme'
import ArtistsSidebar from '../../components/ArtistsSidebar'

const mostRecentAlbums = () => {
  return [
    {
      id: 1,
      title: 'ABC',
      artist: 'ABC',
      artist_slug: 'abc',
      album_slug: 'abc',
      cover_url: 'http://localhost:3000/small_ABC--ABC.jpg'
    },
    {
      id: 2,
      title: 'DEF',
      artist: 'DEF',
      artist_slug: 'def',
      album_slug: 'def',
      cover_url: 'http://localhost:3000/small_DEF--DEF.jpg'
    }
  ]
}

const mostRecentComments = () => {
  return [
    {
      id: 1,
      created_at: '<time>March 8, 2018 2:55am</time>',
      user_name: 'fred',
      user_slug: 'fred',
      gravatar_url: 'https://gravatar.com/avatar/88884e5cb90c02031f031bc',
      name: 'abc',
      path: 'artist/abc'
    },
    {
      id: 2,
      created_at: '<time>March 7, 2018 4:22am</time>',
      user_name: 'jon',
      user_slug: 'jon',
      gravatar_url: 'https://gravatar.com/avatar/88884e5cb90c02031f031bc',
      name: 'def',
      path: 'artist/def/album/def'
    }
  ]
}

describe('<ArtistsSidebar />', () => {
  it('renders without crashing', () => {
    shallow(<ArtistsSidebar mostRecentAlbums={[]} mostRecentComments={[]} />)
  })

  it('renders a sidebar', () => {
    const wrapper = shallow(
      <ArtistsSidebar
        mostRecentAlbums={mostRecentAlbums()}
        mostRecentComments={mostRecentComments()}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
