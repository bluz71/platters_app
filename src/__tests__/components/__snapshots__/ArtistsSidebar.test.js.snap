// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistsSidebar /> renders a sidebar 1`] = `
<aside
  className="ArtistsSidebar"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={2}
  >
    <section
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h4
          className="panel-title"
        >
          New Albums
        </h4>
      </div>
      <ListGroup
        bsClass="list-group"
      >
        <ListGroupItem
          bsClass="list-group-item"
          className="album"
          key="1"
          listItem={false}
        >
          <h5>
            <Link
              replace={false}
              to="/artist/abc/album/abc"
            >
              ABC
            </Link>
          </h5>
          <h5>
            by 
            <Link
              replace={false}
              to="/artist/abc"
            >
              ABC
            </Link>
          </h5>
          <Link
            replace={false}
            to="/artist/abc/album/abc"
          >
            <img
              alt="ABC"
              className="img-responsive center-block"
              src="http://localhost:3000/small_ABC--ABC.jpg"
            />
          </Link>
        </ListGroupItem>
        <ListGroupItem
          bsClass="list-group-item"
          className="album"
          key="2"
          listItem={false}
        >
          <h5>
            <Link
              replace={false}
              to="/artist/def/album/def"
            >
              DEF
            </Link>
          </h5>
          <h5>
            by 
            <Link
              replace={false}
              to="/artist/def"
            >
              DEF
            </Link>
          </h5>
          <Link
            replace={false}
            to="/artist/def/album/def"
          >
            <img
              alt="DEF"
              className="img-responsive center-block"
              src="http://localhost:3000/small_DEF--DEF.jpg"
            />
          </Link>
        </ListGroupItem>
      </ListGroup>
    </section>
    <section
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h4
          className="panel-title"
        >
          New Comments
        </h4>
      </div>
      <ListGroup
        bsClass="list-group"
      >
        <ListGroupItem
          bsClass="list-group-item"
          className="comment"
          key="1"
          listItem={false}
        >
          <h5
            dangerouslySetInnerHTML={
              Object {
                "__html": "<time>March 8, 2018 2:55am</time> by",
              }
            }
          />
          <h5>
            <Link
              replace={false}
              to="/comments/fred"
            >
              fred
            </Link>
          </h5>
          <Link
            replace={false}
            to="/comments/fred"
          >
            <img
              alt="fred"
              className="img-responsive center-block"
              src="https://gravatar.com/avatar/88884e5cb90c02031f031bc"
            />
          </Link>
          <h5>
            in 
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "artist/abc",
                  "state": Object {
                    "scrollToComments": true,
                  },
                }
              }
            >
              abc
            </Link>
          </h5>
        </ListGroupItem>
        <ListGroupItem
          bsClass="list-group-item"
          className="comment"
          key="2"
          listItem={false}
        >
          <h5
            dangerouslySetInnerHTML={
              Object {
                "__html": "<time>March 7, 2018 4:22am</time> by",
              }
            }
          />
          <h5>
            <Link
              replace={false}
              to="/comments/jon"
            >
              jon
            </Link>
          </h5>
          <Link
            replace={false}
            to="/comments/jon"
          >
            <img
              alt="jon"
              className="img-responsive center-block"
              src="https://gravatar.com/avatar/88884e5cb90c02031f031bc"
            />
          </Link>
          <h5>
            in 
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "artist/def/album/def",
                  "state": Object {
                    "scrollToComments": true,
                  },
                }
              }
            >
              def
            </Link>
          </h5>
        </ListGroupItem>
      </ListGroup>
    </section>
  </Col>
</aside>
`;
